File: BMI3/app/src/main/java/com/example/tipper/Quizz.java
package com.example.tipper;

import android.annotation.SuppressLint;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import com.example.tipper.Question;
import com.example.tipper.R;

public class Quizz extends AppCompatActivity
        implements View.OnClickListener {
    // setting up things
    private Button falseButton;
    private Button trueButton;
    private ImageButton nextButton;
    private ImageButton prevButton;
    private ImageView Image;
    private TextView questionTextView;
    private int correct = 0;
    // to keep current Question track
    private int currentQuestionIndex = 0;

    private Question[] questionBank = new Question[] {

            new Question(R.string.a, true),
            new Question(R.string.b, true),
            new Question(R.string.c, true),
            new Question(R.string.d, true),
            new Question(R.string.e, true),
            new Question(R.string.f, true),

    };

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_quizz);

        falseButton = findViewById(R.id.false_button);
        trueButton = findViewById(R.id.true_button);
        nextButton = findViewById(R.id.next_button);
        prevButton = findViewById(R.id.prev_button);

        questionTextView
                = findViewById(R.id.answer_text_view);
        Image = findViewById(R.id.myimage);
        falseButton.setOnClickListener(this);
        trueButton.setOnClickListener(this);
        nextButton.setOnClickListener(this);
        prevButton.setOnClickListener(this);
    }

    @SuppressLint("SetTextI18n")
    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    @Override
    public void onClick(View v)
    {

        switch (v.getId()) {
            case R.id.false_button:
                checkAnswer(false);
                break;

            case R.id.true_button:
                checkAnswer(true);
                break;

            case R.id.next_button:

                if (currentQuestionIndex < 7) {
                    currentQuestionIndex
                            = currentQuestionIndex + 1;

                    if (currentQuestionIndex == 6) {
                        questionTextView.setText(getString(
                                R.string.correct, correct));
                        nextButton.setVisibility(
                                View.INVISIBLE);
                        prevButton.setVisibility(
                                View.INVISIBLE);
                        trueButton.setVisibility(
                                View.INVISIBLE);
                        falseButton.setVisibility(
                                View.INVISIBLE);
                        if (correct > 3)

                            questionTextView.setText(
                                    "Poprawne " + correct
                                            + " "
                                            + "Z 6");

                        else
                            Image.setImageResource(
                                    R.drawable.olej);
                    }
                    else {
                        updateQuestion();
                    }
                }

                break;
            case R.id.prev_button:
                if (currentQuestionIndex > 0) {
                    currentQuestionIndex
                            = (currentQuestionIndex - 1)
                            % questionBank.length;
                    updateQuestion();
                }
        }
    }

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    private void updateQuestion()
    {
        Log.d("Current",
                "onClick: " + currentQuestionIndex);

        questionTextView.setText(
                questionBank[currentQuestionIndex]
                        .getAnswerResId());
        // setting the textview with new Question
        switch (currentQuestionIndex) {
            case 1:
                // setting up image for each
                // Question
                Image.setImageResource(R.drawable.olej);
                break;
            case 2:
                Image.setImageResource(R.drawable.left);
                break;
            case 3:
                Image.setImageResource(R.drawable.right);
                break;
            case 4:
                Image.setImageResource(R.drawable.abc_vector_test);
                break;
            case 5:
                Image.setImageResource(R.drawable.ic_launcher_background);
                break;
            case 6:
                Image.setImageResource(R.drawable.dieta1);
                break;
            case 7:
                Image.setImageResource(R.drawable.ic_launcher_foreground);
                break;
        }
    }
    private void checkAnswer(boolean userChooseCorrect)
    {
        boolean answerIsTrue
                = questionBank[currentQuestionIndex]
                .isAnswerTrue();
        int toastMessageId;

        if (userChooseCorrect == answerIsTrue) {
            toastMessageId = R.string.correct_answer;
            correct++;
        }
        else {
            // showing toast
            // message correct
            toastMessageId = R.string.wrong_answer;
        }
        Toast.makeText(Quizz.this, toastMessageId,
                Toast.LENGTH_SHORT).show();
    }
}

File: Kotlin_Final/app/src/main/java/com/example/tipper/Quizz.kt
package com.example.tipper

import android.annotation.SuppressLint
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.*
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity

class Quizz : AppCompatActivity(), View.OnClickListener {
    // setting up things
    private var falseButton: Button? = null
    private var trueButton: Button? = null
    private var nextButton: ImageButton? = null
    private var prevButton: ImageButton? = null
    private var Image: ImageView? = null
    private var questionTextView: TextView? = null
    private var correct = 0

    // to keep current Question track
    private var currentQuestionIndex = 0
    private val questionBank = arrayOf(
        Question(R.string.a, true),
        Question(R.string.b, true),
        Question(R.string.c, true),
        Question(R.string.d, true),
        Question(R.string.e, true),
        Question(R.string.f, true)
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quizz)
        falseButton = findViewById<Button>(R.id.false_button)
        trueButton = findViewById<Button>(R.id.true_button)
        nextButton = findViewById<ImageButton>(R.id.next_button)
        prevButton = findViewById<ImageButton>(R.id.prev_button)
        questionTextView = findViewById(R.id.answer_text_view)
        Image = findViewById(R.id.myimage)

    }

    @SuppressLint("SetTextI18n")
    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    override fun onClick(v: View) {
        when (v.id) {
            R.id.false_button -> checkAnswer(false)
            R.id.true_button -> checkAnswer(true)
            R.id.next_button -> if (currentQuestionIndex < 7) {
                currentQuestionIndex += 1
                if (currentQuestionIndex == 6) {
                    questionTextView!!.text = getString(
                        R.string.correct, correct
                    )
                    nextButton!!.visibility = View.INVISIBLE
                    prevButton!!.visibility = View.INVISIBLE
                    trueButton!!.visibility = View.INVISIBLE
                    falseButton!!.visibility = View.INVISIBLE
                    if (correct > 3) questionTextView!!.text = ("Poprawne " + correct
                            + " "
                            + "Z 6") else Image!!.setImageResource(
                        R.drawable.olej
                    )
                } else {
                    updateQuestion()
                }
            }
            R.id.prev_button -> if (currentQuestionIndex > 0) {
                currentQuestionIndex = ((currentQuestionIndex - 1)
                        % questionBank.size)
                updateQuestion()
            }
        }
    }

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    private fun updateQuestion() {
        Log.d(
            "Current",
            "onClick: $currentQuestionIndex"
        )
        questionTextView?.setText(
            questionBank[currentQuestionIndex]
                .answerResId
        )
        when (currentQuestionIndex) {
            1 ->

                Image!!.setImageResource(R.drawable.olej)
            2 -> Image!!.setImageResource(R.drawable.left)
            3 -> Image!!.setImageResource(R.drawable.right)
            4 -> Image!!.setImageResource(R.drawable.abc_vector_test)
            5 -> Image!!.setImageResource(R.drawable.ic_launcher_background)
            6 -> Image!!.setImageResource(R.drawable.dieta1)
            7 -> Image!!.setImageResource(R.drawable.ic_launcher_foreground)
        }
    }

    private fun checkAnswer(userChooseCorrect: Boolean) {
        val answerIsTrue = questionBank[currentQuestionIndex]
            .isAnswerTrue
        val toastMessageId: Int
        if (userChooseCorrect == answerIsTrue) {
            toastMessageId = R.string.correct_answer
            correct++
        } else {
            // showing toast
            // message correct
            toastMessageId = R.string.wrong_answer
        }
        Toast.makeText(
            this@Quizz, toastMessageId,
            Toast.LENGTH_SHORT
        ).show()
    }
}